data_model/src/isi.rs

```diff
@@ -79,790 +71,1151 @@ pub mod model {
+    /// Enum with all supported [`RemoveKeyValue`] instructions.
+    pub enum RemoveKeyValueBox {
+        /// Remove key value from [`Domain`].
+        Domain(RemoveKeyValue<Domain>),
+        /// Remove key value from [`Account`].
+        Account(RemoveKeyValue<Account>),
+        /// Remove key value from [`AssetDefinition`].
+        AssetDefinition(RemoveKeyValue<AssetDefinition>),
+        /// Remove key value from [`Asset`].
+        Asset(RemoveKeyValue<Asset>),
     }
 }
 
-impl ConditionalExpr {
-    /// Construct [`If`].
-    pub fn new<C: Into<EvaluatesTo<bool>>, T: Into<InstructionExpr>>(
-        condition: C,
-        then: T,
-    ) -> Self {
-        Self {
-            condition: condition.into(),
-            then: then.into(),
-            otherwise: None,
-        }
-    }
-    /// [`If`] constructor with `Otherwise` instruction.
-    pub fn with_otherwise<
-        C: Into<EvaluatesTo<bool>>,
-        T: Into<InstructionExpr>,
-        O: Into<InstructionExpr>,
-    >(
-        condition: C,
-        then: T,
-        otherwise: O,
-    ) -> Self {
-        Self {
-            condition: condition.into(),
-            then: then.into(),
-            otherwise: Some(otherwise.into()),
-        }
+isi_box! {
+    /// Enum with all supported [`Register`] instructions.
+    pub enum RegisterBox {
+        /// Register [`Peer`].
+        Peer(Register<Peer>),
+        /// Register [`Domain`].
+        Domain(Register<Domain>),
+        /// Register [`Account`].
+        Account(Register<Account>),
+        /// Register [`AssetDefinition`].
+        AssetDefinition(Register<AssetDefinition>),
+        /// Register [`Asset`].
+        Asset(Register<Asset>),
+        /// Register [`Role`].
+        Role(Register<Role>),
+        /// Register [`Trigger`].
+        Trigger(Register<Trigger<TriggeringFilterBox>>)
     }
 }
 
-impl Fail {
-    /// Construct [`Fail`].
-    pub fn new(message: &str) -> Self {
-        Self {
-            message: String::from(message),
-        }
+isi_box! {
+    /// Enum with all supported [`Unregister`] instructions.
+    pub enum UnregisterBox {
+        /// Unregister [`Peer`].
+        Peer(Unregister<Peer>),
+        /// Unregister [`Domain`].
+        Domain(Unregister<Domain>),
+        /// Unregister [`Account`].
+        Account(Unregister<Account>),
+        /// Unregister [`AssetDefinition`].
+        AssetDefinition(Unregister<AssetDefinition>),
+        /// Unregister [`Asset`].
+        Asset(Unregister<Asset>),
+        /// Unregister [`Role`].
+        Role(Unregister<Role>),
+        /// Unregister [`Trigger`].
+        Trigger(Unregister<Trigger<TriggeringFilterBox>>)
     }
 }
 
-impl SetParameterExpr {
-    /// Construct [`SetParameterExpr`].
-    pub fn new<P: Into<EvaluatesTo<Parameter>>>(parameter: P) -> Self {
-        Self {
-            parameter: parameter.into(),
-        }
+isi_box! {
+    /// Enum with all supported [`Mint`] instructions.
+    pub enum MintBox {
```

Comment:
do you think we need these `*Box` structs other than `InstructionBox`. Would it make sense to inline it all to `InstructionBox`. What I'm asking is, do we have to group all `Mint` instructions under `MintBox` or can we just make them variants of `InstructionBox` directly

Reply:
maybe we should? flat representation will also take less memory

Reply:
I also thought of such approach

Reply:
Maybe...

The only downside I can see (except repeating instruction names) is that in the Executor there will be no simple way to deny all registrations for example, as will be no `visit_register(RegisterBox)`. Same for all other boxes

Reply:
ok, let's leave this as is then